File: app/src/main/java/com/esgi/greenrepack/ProjectAdapter.java
package com.esgi.greenrepack;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Adapter;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.esgi.greenrepack.obj.Association;
import com.esgi.greenrepack.obj.Project;

import org.jetbrains.annotations.NotNull;

import java.util.ArrayList;
import java.util.List;

public class ProjectAdapter extends RecyclerView.Adapter<ProjectAdapter.MyViewHolder> {
    @NonNull
    @NotNull

    List<Project> projectListII = new ArrayList<>();
    List<Project> ProjectListIII;
    List<Project> ProjectListIV;

    int ProjectList[] = {R.drawable.travauxmaritime, R.drawable.ponttillionville,R.drawable.traintravaux};

    List<String> Labels;
    String LabelsX[];
    Context context;

    public ProjectAdapter(int[] projectList, String[] labels, Context context) {
        ProjectList = projectList;
        LabelsX = labels;
        this.context = context;
    }

    /*public ProjectAdapter(List<Project> projectListII, List<String> labels, Context context) {
        this.projectListII = projectListII;
        Labels = labels;
        this.context = context;
    }*/

    public ProjectAdapter.MyViewHolder onCreateViewHolder(@NonNull @NotNull ViewGroup parent, int viewType) {

        LayoutInflater inflater = LayoutInflater.from(context);
        View view = inflater.inflate(R.layout.project_label_row,parent,false);
        return new ProjectAdapter.MyViewHolder(view);

    }

    @Override
    public void onBindViewHolder(@NonNull @NotNull ProjectAdapter.MyViewHolder holder, int position) {

        holder.ProjectLabel.setText(projectListII.get(position).getLibelle());
        holder.Projectpicture.setImageResource(ProjectList[position]);

    }

    @Override
    public int getItemCount() {
        return ProjectList.length;
    }

    public class MyViewHolder extends RecyclerView.ViewHolder {

        TextView ProjectLabel;
        ImageView Projectpicture;



        public MyViewHolder(@NonNull View itemView) {
            super(itemView);

            ProjectLabel = itemView.findViewById(R.id.ProjectLabel);
            Projectpicture = itemView.findViewById(R.id.ProjectThumbs);
        }
    }

}

File: app/src/main/java/com/esgi/greenrepack/ProjectAdapter.kt
package com.esgi.greenrepack

import android.content.Context
import android.content.Intent
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.bumptech.glide.Glide
import com.esgi.greenrepack.ProjectAdapter.MyViewHolder
import com.esgi.greenrepack.obj.Project
import kotlinx.android.synthetic.main.project_label_row.view.*

class ProjectAdapter(private val projects: List<Project>, private val context: Context) : RecyclerView.Adapter<MyViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MyViewHolder {
        val inflater = LayoutInflater.from(parent.context)
        val view = inflater.inflate(R.layout.project_label_row, parent, false)
        return MyViewHolder(view)
    }

    override fun onBindViewHolder(holder: MyViewHolder, position: Int) {
        holder.itemView.setOnClickListener {
            val intent = Intent(context, ProjectActivity::class.java)
            //intent.putExtra("film_id",film.id)
            context.startActivity(intent)
        }
        holder.projectLabel.text = projects[position].libelle
        Glide.with(context)
            .load(projects[position].logo)
            .placeholder(R.drawable.ic_launcher_background)
            .into(holder.projectpicture)
    }

    override fun getItemCount(): Int {
        return projects.size
    }

    inner class MyViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        var projectLabel: TextView = itemView.project_label
        var projectpicture: ImageView = itemView.project_image
    }
}